### PR TITLE
Retry quickly when references changed

### PR DESCRIPTION
**Pull request by dims**
Lets not wait 4 hours and retry quickly. If this works, we can add another parameter instead of hardcoding 5 mins.

Change-Id: I2ab92772f417124c4684bb5fb17bc5437923812d

### CODE REVIEW

File: cmd/publishing-bot/main.go [hunk 1]
```diff
@@ -183,6 +185,11 @@ func main() {
 					githubIssueErrorf("Failed to report logs on github issue: %v", err)
 					server.SetHealth(false, hash)
 				}
+				if strings.HasSuffix(err.Error(), memory.ErrRefHasChanged.Error()) {
+					// If the issue is just "reference has changed concurrently",
+					// then let us wait for 5 minutes and try again.
```
Comment (nikhita) [inline cmd/publishing-bot/main.go]: Let's also add some logging here...also, can we add a comment here with a TODO to fix the actual flake?

Comment (nikhita) [reply cmd/publishing-bot/main.go]: Can this end up in some weird recursion? :upside_down_face: 

We have seen this error consecutively several times...maybe we should let it retry only a certain number of times?

Comment (dims) [reply cmd/publishing-bot/main.go]: done.

Comment (dims) [reply cmd/publishing-bot/main.go]: i was thinking we should try this as-is and we can remove this if it does not work.

**Comment by nikhita** [comment]
Why do we have travis CI run here? It was removed in https://github.com/kubernetes/publishing-bot/pull/128.....

**Comment by nikhita** [comment]
We have prow now which runs `make` so I'll remove the webhook for travis.

https://github.com/kubernetes/test-infra/blob/a967141fc1ac3faee4e703dcf8ba4201f17c9cfa/config/jobs/kubernetes/publishing-bot/publishing-bot-presubmits.yaml#L2-L11

**Comment by nikhita** [comment]
Closing and reopening to remove travis.

**Comment by nikhita** [comment]
O_o

**Comment by nikhita** [comment]
Travis didn't run any of the past few PRs...why this one. thinking:

**Pull request merged**